src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEnginePublisher.java

```diff
@@ -220,10 +260,36 @@ public FormValidation doCheckManifestPattern(@QueryParameter String value) {
       if (Strings.isNullOrEmpty(value)) {
         return FormValidation.error(Messages.KubernetesEnginePublisher_ManifestRequired());
       }
-
       return FormValidation.ok();
     }
 
+    public ListBoxModel doFillZoneItems(
+        @AncestorInPath Jenkins context,
+        @QueryParameter("projectId") final String projectId,
+        @QueryParameter("credentialsId") final String credentialsId) {
+      ListBoxModel items = new ListBoxModel();
+      items.add("- none -", "");
+      if (!Strings.isNullOrEmpty(projectId) && !Strings.isNullOrEmpty(credentialsId)) {
+        try {
+          ComputeClient compute = getComputeClient(context, credentialsId);
+          List<Zone> zones = compute.getZones(projectId);
+
+          // This enables auto-populating the zone when there are zones.
+          if (!zones.isEmpty()) {
+            items.clear();
+          }
+
+          for (Zone z : zones) {
+            items.add(z.getName());
+          }
+        } catch (IOException ioe) {
+          items.clear();
+          items.add(ZONE_FILL_ERROR, "");
+        }
+      }
+      return items;
+    }
+
     public FormValidation doCheckZone(@QueryParameter String value) {
```

Comment:
Now that a compute client is available, do you think it would make since to add some verification logic to this method?

Reply:
Yes. Perhaps I would do it only if the entry mode is "textbox".

Reply:
SGTM

Reply:
Resolving this as this will be implemented in the next PR.